style/javascript.md

```diff
@@ -552,3 +553,52 @@ Yes:
 ```js
 $(".some-class span").hide();
 ```
+
+#### Don't use Underscore
+
+We use ES6/7 which includes many of the features of Underscore.js! Using Underscore should be avoided in favor of these native language features.
+
+There are a couple of methods that are sufficiently complicated and don't have a direct equivalent so instead we have a [custom-built](https://lodash.com/custom-builds) copy of [lodash](https://lodash.com/) containing only those specific methods. You can find this file at: `third_party/javascript-khansrc/lodash/lodash.js` along with instructions on how to build it and exactly what methods are included.
+
+What follows is a method-by-method set of equivalents for what Underscore provides and what you could be using in ES6/7 instead:
+
+Method | Use...                                | ...instead of
+--------- | ------------------------------------- | ----------------------
+bind | `fn.bind(someObj, args)` | `_.bind(fn, someObj, args)`
+bind | `(a, b) => { ... }` <sup>[1](#u1)</sup> | `_.bind(function(a, b) { ... }, this)`
+bindAll | `obj.method = obj.method.bind(someObj);` <sup>[2](#u2)</sup> | `_.bindAll(someObj, "method")`
+clone | No alternative at the moment! <sup>[3](#u3)</sup> |
+debounce | Our custom lodash build. |
+defer | `setTimeout(fn, 0);` | `_.defer(fn);`
+delay | `setTimeout(fn, 2000);` | `_.delay(fn, 2000);`
+each (array) | `array.forEach((val, i) => {})` | `_.each(array, (val, i) => {})`
+each (array) | `for (const val of array) {}` | `_.each(array, fn)`
+each (object) | `for (const [key, val] of Object.entries(obj)) {}` | `_.each(obj, fn)`
+extend (new) | `{...options, prop: 1}` | `_.extend({}, options, {prop: 1})`
+extend (assign) | `Object.assign(json, this.model.toJSON())` | `_.extend(json, this.model.toJSON())`
+filter | `array.filter(checkFn)` | `_.filter(array, checkFn)`
+has (array) | `array.includes(value)` | `_.has(array, value)`
+has (object) | `obj.hasOwnProperty(value)` <sup>[4](#u4)</sup> | `_.has(obj, value)`
+isArray | `Array.isArray(someObj)` | `_.isArray(someObj)`
+isFunction | `typeof fn === "function"` | `_.isFunction(fn)`
+isString | `typeof obj === "string"` | `_.isString(obj)`
+keys | `Object.keys(obj)` | `_.keys(obj)`
+last | `someArray[someArray.length - 1]` <sup>[5](#u5)</sup> | `_.last(someArray)`
+map | `array.map(mapFn)` | `_.map(array, mapFn)`
```

Comment:
Lodash and Underscore allow callback shorthands like `_.map(array, 'id)` instead of:
```js
array.map(function(v) { return v != null && v.id; });
```